Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -64,5 +64,9 @@ public record DafnyDocument(
     }
 
     public int LinesCount => VerificationTree.Range.End.Line;
+
+    public static int NumberOfLines(string text) {
+      return text.Count(c => c == '\n') + 1;
```

Comment:
I don't think we should compute a number of lines from the full text anywhere besides in `TextChangeProcessor`. Instead of working with `TextDocumentItem` we can work with our own datatype (like `DocumentTextBuffer`) that besides storing the text as a `string` also stores the number of lines.

In the future we'll change `TextChangeProcessor` so it'll update `DocumentTextBuffer` with a cost linear to the amount of changes (instead of the the current cost which is linear in the document size). I'm not sure what the text buffer data-structure will look like, but maybe this can inspire us: https://code.visualstudio.com/blogs/2018/03/23/text-buffer-reimplementation

Reply:
I implemented your suggestion, I like it ! Makes more sense than in the DafnyDocument.